tests/utils/test_arguments.py

```diff
@@ -0,0 +1,85 @@
+from dataclasses import FrozenInstanceError
+
+import pytest
+import pytorch_lightning as pl
+
+from pl_bolts.utils.arguments import LightningArgumentParser, LitArg, gather_lit_args
+
+
+class DummyParentModel(pl.LightningModule):
+
+    name = "parent-model"
+
+    def __init__(self, a: int, b: str, c: str = "parent_model_c"):
+        super().__init__()
+        self.save_hyperparameters()
+
+    def forward(self, x):
+        pass
+
+
+class DummyParentDataModule(pl.LightningDataModule):
+
+    name = "parent-dm"
+
+    def __init__(self, d: str, c: str = "parent_dm_c"):
+        super().__init__()
+        self.d = d
+        self.c = c
+
+
+def test_lightning_argument_parser():
+    parser = LightningArgumentParser(ignore_required_init_args=False)
+    assert parser.ignore_required_init_args is False
+    parser = LightningArgumentParser(ignore_required_init_args=True)
+    assert parser.ignore_required_init_args is True
+
+
+def test_lit_arg_immutable():
+    arg = LitArg("some_arg", (int,), 1, False)
+    with pytest.raises(FrozenInstanceError):
+        arg.default = 0
+    assert arg.default == 1
+
+
+@pytest.mark.parametrize(
+    "obj,expected",
+    [
+        pytest.param(
+            DummyParentModel,
+            {"a": (int, None, True), "b": (str, None, True), "c": (str, "parent_model_c", False),},
+            id="dummy-parent-model",
+        ),
+        pytest.param(
+            DummyParentDataModule, {"d": (str, None, True), "c": (str, "parent_dm_c", False)}, id="dummy-parent-dm",
+        ),
+    ],
+)
+def test_gather_lit_args(obj, expected):
+    lit_args = gather_lit_args(obj)
+    assert len(lit_args) == len(expected)
+    for lit_arg, (k, v) in zip(lit_args, expected.items()):
+        assert lit_arg.name == k
+        assert lit_arg.types[0] == v[0]
+        assert lit_arg.default == v[1]
+        assert lit_arg.required == v[2]
+
```

Comment:
Add a test for user giving an arg that doesnt exist?

Reply:
sure

Reply:
Added simple one